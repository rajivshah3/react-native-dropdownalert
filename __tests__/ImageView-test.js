import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImageView from '../imageview';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import errorImage from '../assets/error.png';

test('renders imageview with uri source', () => {
  const wrapper = shallow(<ImageView source={'https://facebook.github.io/react/img/logo_og.png'} />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders imageview with number source', () => {
  const wrapper = shallow(<ImageView source={errorImage} />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders imageview without source', () => {
  const wrapper = shallow(<ImageView source={null} />);
  const tree = toJson(wrapper);
  expect(tree).toEqual("");
});
test('renders imageview with style and source', () => {
  const wrapper = shallow(<ImageView style={{ width: 44, height: 44 }} source={'https://facebook.github.io/react/img/logo_og.png'} />);
  const tree = toJson(wrapper);
  expect(tree.props.style).toBeDefined();
});
test('renders imageview to have width and height without it defined in style', () => {
  const wrapper = shallow(<ImageView style={{ flex: 1 }} source={'https://facebook.github.io/react/img/logo_og.png'} />);
  const tree = toJson(wrapper);
  expect(tree.props.style.width).toBe(36);
  expect(tree.props.style.height).toBe(36);
});
