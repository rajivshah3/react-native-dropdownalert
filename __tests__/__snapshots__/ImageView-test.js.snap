// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders imageview with number source 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../assets/error.png",
    }
  }
  style={
    Object {
      "alignSelf": "center",
      "height": 36,
      "padding": 8,
      "width": 36,
    }
  }
/>
`;

exports[`renders imageview with uri source 1`] = `
<Image
  source={
    Object {
      "uri": "https://facebook.github.io/react/img/logo_og.png",
    }
  }
  style={
    Object {
      "alignSelf": "center",
      "height": 36,
      "padding": 8,
      "width": 36,
    }
  }
/>
`;
